.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,89 @@
+name: Build and run tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - develop
+
+jobs:
+  build-and-test:
+    runs-on: ${{ matrix.platform }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install stable toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+
+    - name: Get current date
+      run: echo "CURRENT_DATE=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+
```

Comment:
```suggestion
      if: matrix.platform == 'macos-latest' || matrix.platform == 'ubuntu-latest'
      
    - name: Get current date
      if: matrix.platform == 'windows-latest'
      run: echo "CURRENT_DATE=$(Get-Date -Format "yyyy-MM-dd")" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8
```

Reply:
what's this change?

Reply:
This gets the current date on Windows (for the cache key) since the command is different

Reply:
oh I misread it lol thanks